src/components/TokenSelector.tsx

```diff
@@ -88,7 +90,7 @@ const SelectedTokenWrapper = styled.span`
   }
 `
 
-function renderOptionLabel(token: TokenDetails): React.ReactNode {
+function renderOptionLabel(token: TokenDetails | TokenBalanceDetails): React.ReactNode {
```

Comment:
I don't particularly like how we don't know at this point if token is one type or another
Maybe we should always deal with `TokenBalanceDetails`
The problem stems from `TradeWidget/index.tsx::L367`

Reply:
Yes, I thought about using throughout `TokenBalanceDetails`, but this type requires things related to user balance, which are not available when logged out.
Since we do want to display the token selector even when not logged in, I opted for the current version.